consensus/src/main/java/org/apache/iotdb/consensus/ratis/SnapshotStorage.java

```diff
@@ -141,8 +142,14 @@ public SnapshotInfo getLatestSnapshot() {
     }
     TermIndex snapshotTermIndex = Utils.getTermIndexFromDir(latestSnapshotDir);
 
+    List<File> actualSnapshotFiles = applicationStateMachine.getSnapshotFiles(latestSnapshotDir);
+    List<Path> filesIncludedInSnapshot =
+        actualSnapshotFiles != null
+            ? actualSnapshotFiles.stream().map(File::toPath).collect(Collectors.toList())
+            : getAllFilesUnder(latestSnapshotDir);
+
```

Comment:
Can we put all these logics into `applicationStateMachine.getSnapshotFiles`, in that method we encapsulate all details about where to get the snapshot files, from logFile or just listing files in a directory. In fact, I find that there already exists such logics in `DataRegionStateMachine.getSnapshotFiles`.

BTW, you can change the interface to directly return `List<Path>` instead of `List<File>`.